src/matrix/matrixtdist.jl

```diff
@@ -0,0 +1,196 @@
+MatrixTDist(ν::Real, M::AbstractMatrix, Σ::LinearAlgebra.Cholesky, Ω::LinearAlgebra.Cholesky) = MatrixTDist(ν, M, PDMat(Σ), PDMat(Ω))
+MatrixTDist(ν::Real, M::AbstractMatrix, Σ::AbstractPDMat,          Ω::AbstractMatrix)         = MatrixTDist(ν, M, Σ,        PDMat(Ω))
+MatrixTDist(ν::Real, M::AbstractMatrix, Σ::AbstractPDMat,          Ω::LinearAlgebra.Cholesky) = MatrixTDist(ν, M, Σ,        PDMat(Ω))
+
+#  -----------------------------------------------------------------------------
+#  REPL display
+#  -----------------------------------------------------------------------------
+
+ show(io::IO, d::MatrixTDist) = show_multline(io, d, [(:ν, d.ν), (:M, d.M), (:Σ, Matrix(d.Σ)), (:Ω, Matrix(d.Ω))])
+
+#  -----------------------------------------------------------------------------
+#  Conversion
+#  -----------------------------------------------------------------------------
+
+function convert(::Type{MatrixTDist{T}}, d::MatrixTDist) where T <: Real
+    MM = convert(AbstractArray{T}, d.M)
+    ΣΣ = convert(AbstractArray{T}, d.Σ)
+    ΩΩ = convert(AbstractArray{T}, d.Ω)
+    MatrixTDist{T, typeof(MM), typeof(ΣΣ)}(T(d.ν), MM, ΣΣ, ΩΩ, T(d.logc0))
+end
+
+function convert(::Type{MatrixTDist{T}}, ν, M::AbstractMatrix, Σ::AbstractPDMat, Ω::AbstractPDMat, logc0) where T <: Real
+    MM = convert(AbstractArray{T}, M)
+    ΣΣ = convert(AbstractArray{T}, Σ)
+    ΩΩ = convert(AbstractArray{T}, Ω)
+    MatrixTDist{T, typeof(MM), typeof(ΣΣ)}(T(ν), MM, ΣΣ, ΩΩ, T(logc0))
+end
+
+#  -----------------------------------------------------------------------------
+#  Properties
+#  -----------------------------------------------------------------------------
+
+size(d::MatrixTDist) = size(d.M)
+
+rank(d::MatrixTDist) = minimum( size(d) )
+
+insupport(d::MatrixTDist, X::Matrix) = isreal(X) && size(X) == size(d)
+
+function mean(d::MatrixTDist)
+
+  n, p = size(d)
+
+  (d.ν + p - n > 1) ? (return d.M) : throw(ArgumentError("mean only defined for df + p - n > 1"))
+
+end
+
+mode(d::MatrixTDist) = d.M
+
+params(d::MatrixTDist) = (d.ν, d.M, d.Σ, d.Ω)
+
+@inline partype(d::MatrixTDist{T}) where {T <: Real} = T
+
+#  -----------------------------------------------------------------------------
+#  Evaluation
+#  -----------------------------------------------------------------------------
+
+function _matrixtdist_logc0(Σ::AbstractPDMat, Ω::AbstractPDMat, ν::Real)
+    #  returns the natural log of the normalizing constant for the pdf
+
+    n = dim(Σ)
+    p = dim(Ω)
+
+    term1 = logmvgamma(p, (ν + n + p - 1) / 2)
+    term2 = - (n * p / 2) * logπ
+    term3 = - logmvgamma(p, (ν + p - 1) / 2)
+    term4 = (-n / 2) * logdet(Ω)
+    term5 = (-p / 2) * logdet(Σ)
+
+    term1 + term2 + term3 + term4 + term5
+
+end
+
+function logkernel(d::MatrixTDist, X::AbstractMatrix)
+
+    n, p = size(d)
+    A = X - d.M
+
+    (-(d.ν + n + p - 1) / 2) * logdet( I + (d.Σ \ A) * (d.Ω \ A') )
+
+end
+
+_logpdf(d::MatrixTDist, X::AbstractMatrix) = logkernel(d, X) + d.logc0
```

Comment:
Why `_logpdf` and not `logpdf`?

Reply:
That would appear to be the standard? See [here](https://github.com/JuliaStats/Distributions.jl/blob/master/src/matrix/wishart.jl#L103), [here](https://juliastats.github.io/Distributions.jl/latest/matrix/#Internal-Methods-(for-creating-your-own-matrix-variate-distributions)-1), and [here](https://juliastats.github.io/Distributions.jl/latest/extends/#Create-a-Matrix-variate-Distribution-1). 

Matrix-variates have a generic `logpdf` fallback [here](https://github.com/JuliaStats/Distributions.jl/blob/d876df09f06a8d559e4733a875fbd11709f2dc60/src/matrixvariates.jl#L102) that checks dims. 

(While we're on the subject, I guess it might make more sense for the fallback to call `insupport` instead of just checking the dims, but that's a PR for another day...)